plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/data/ExecuteInsertBatchImpl.java

```diff
@@ -159,54 +166,77 @@ StringBuilder prepareQueryForStatement(
             query.append(attributeName);
         }
         query.append(")\n\tVALUES "); //$NON-NLS-1$
+        
+        // Filtered attributes to remove columns that are ignored
+        DBSAttributeBase[] filteredAttributes = null;
+        DBDValueHandler[] filteredHandlers = null;
+        Object[] filteredAttributeValues = null;
+
+        if (skipAttributes.size() > 0) {
+            List<DBSAttributeBase> attributesList = new ArrayList<DBSAttributeBase>(Arrays.asList(attributes));
+            filteredAttributes = 
+                    IntStream.range(0, attributesList.size())
+                            .filter(i -> !skipAttributes.contains(i))
+                            .mapToObj(attributesList::get)
+                            .collect(Collectors.toList())
+                            .stream().toArray(DBSAttributeBase[]::new);
+
+            List<DBDValueHandler> handlersList = new ArrayList<DBDValueHandler>(Arrays.asList(handlers));
+            filteredHandlers = 
+                    IntStream.range(0, handlersList.size())
+                            .filter(i -> !skipAttributes.contains(i))
+                            .mapToObj(handlersList::get)
+                            .collect(Collectors.toList())
+                            .stream().toArray(DBDValueHandler[]::new);
+            
+            List<Object> attributeValuesList = new ArrayList<Object>(Arrays.asList(attributeValues));
+            filteredAttributeValues = 
+                    IntStream.range(0, attributeValuesList.size())
+                            .filter(j -> !skipAttributes.contains(j % attributes.length))
+                            .mapToObj(attributeValuesList::get)
+                            .collect(Collectors.toList())
+                            .stream().toArray();
```

Comment:
We have indexes of used attributes, what's the point of this list, anyway? Using some math we can calculate the desired index based on current attribute.

Pseudocode:
```java
for (int i = 0; i < attributes.length; i++) {
    for (int j = 0; j < values.length / attributes.length; j++) {
        var value = values[i * attributes.length + j];
    }
}

Reply:
Yeah removed all filtered fields.